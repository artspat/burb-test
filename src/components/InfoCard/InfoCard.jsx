import React from 'react';
import cx from 'classnames';
import { Rating } from '../Rating';
import './InfoCard.scss';

export const InfoCard = ({ className = '', title, values = [], rating = -1 }) => {
    return (
        <div className={cx('b-info-card', className)}>
            <h3 className="b-info-card__title">{title}</h3>
            {values.length && values.map((item, index) => (
                <div key={`${item.title}_${index}`} className="b-info-card__item">
                    <span className="b-info-card__item-title">{item.title}:&nbsp;</span>
                    <span>{item.value}</span>
                </div>
            ))}
            {rating !== -1 && (
                <div className="b-info-card__rating">
                    <Rating value={rating} />
                </div>
            )}
        </div>
    );
};
