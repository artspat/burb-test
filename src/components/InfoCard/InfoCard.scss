@import '../../styles/global.scss';

.b-info-card {
    padding: 10px;
    border-radius: 7px;
    background-color: $grey;

    &__title {
        margin: 0;
        margin-bottom: 10px;
        font-size: 24px;
        font-weight: normal;
    }

    &__item {
        padding: 3px 5px;

        &-title {
            font-weight: bold;
        }
    }

    &__rating {
        margin-top: 15px;
    }
}
