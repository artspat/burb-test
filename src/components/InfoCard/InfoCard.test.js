import React from 'react';
import { shallow } from 'enzyme';
import { InfoCard } from './InfoCard';

describe('<InfoCard />', () => {
    const info = [
        { title: '1', value: 1 },
        { title: '2', value: 2 },
    ];

    describe('when props are passed', () => {
        it('should render component', () => {
            const wrapper = shallow(<InfoCard title="Info cart" values={info} rating={2} className="css-class" />);
            expect(wrapper).toMatchSnapshot();
        });
    });
    
    describe('when no values passed', () => {
        it('should not render values', () => {
            const wrapper = shallow(<InfoCard title="Info cart" rating={2} />);
            expect(wrapper).toMatchSnapshot();
        });
    });
    
    describe('when no rating passed', () => {
        it('should not render rating', () => {
            const wrapper = shallow(<InfoCard title="Info cart" values={info} />);
            expect(wrapper).toMatchSnapshot();
        });
    });
});
