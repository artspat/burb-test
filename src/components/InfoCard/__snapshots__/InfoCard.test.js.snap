// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InfoCard /> when no rating passed should not render rating 1`] = `
<div
  className="b-info-card"
>
  <h3
    className="b-info-card__title"
  >
    Info cart
  </h3>
  <div
    className="b-info-card__item"
    key="1_0"
  >
    <span
      className="b-info-card__item-title"
    >
      1
      : 
    </span>
    <span>
      1
    </span>
  </div>
  <div
    className="b-info-card__item"
    key="2_1"
  >
    <span
      className="b-info-card__item-title"
    >
      2
      : 
    </span>
    <span>
      2
    </span>
  </div>
</div>
`;

exports[`<InfoCard /> when no values passed should not render values 1`] = `
<div
  className="b-info-card"
>
  <h3
    className="b-info-card__title"
  >
    Info cart
  </h3>
  0
  <div
    className="b-info-card__rating"
  >
    <Rating
      value={2}
    />
  </div>
</div>
`;

exports[`<InfoCard /> when props are passed should render component 1`] = `
<div
  className="b-info-card css-class"
>
  <h3
    className="b-info-card__title"
  >
    Info cart
  </h3>
  <div
    className="b-info-card__item"
    key="1_0"
  >
    <span
      className="b-info-card__item-title"
    >
      1
      : 
    </span>
    <span>
      1
    </span>
  </div>
  <div
    className="b-info-card__item"
    key="2_1"
  >
    <span
      className="b-info-card__item-title"
    >
      2
      : 
    </span>
    <span>
      2
    </span>
  </div>
  <div
    className="b-info-card__rating"
  >
    <Rating
      value={2}
    />
  </div>
</div>
`;
